Author: Sergei Golovan <sgolovan@debian.org>
Description: This dirty hack fixes segfaults if Tcl is built with stubs
 and Expect is used directly from C program.
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=588817
Example:
 #include <stdio.h>
 #include <tcl8.5/expect.h>
 int main()
 {
     FILE *pipe;
     char *some_command = "uname";
     char datum;
     pipe = exp_popen(some_command);
     if (pipe == NULL) return 1;
     while ((datum = getc (pipe)) != EOF)
 	printf("%c",datum);
 }

--- expect-5.45.orig/exp_clib.c
+++ expect-5.45/exp_clib.c
@@ -117,7 +117,11 @@
 #include <stdlib.h>		/* for malloc */
 #endif
 
-#include <tcl.h>
+#define ckalloc(x) Tcl_Alloc(x)
+#define ckfree(x) Tcl_Free(x)
+extern char *Tcl_ErrnoMsg(int err);
+extern char *Tcl_Alloc(unsigned int size);
+extern void Tcl_Free(char *ptr);
 #include "expect.h"
 #define TclRegError exp_TclRegError
 
